src/test/java/com/api/UserTests.java

```diff
@@ -3,67 +3,68 @@
 import java.net.http.HttpResponse;
 import java.util.Map;
 
-import com.api.constants.JsonConstant;
-import com.api.services.UserService;
-import com.api.utils.TemplateUtils;
+import com.api.utils.RequestBodyBuilderUtil;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
 import com.api.models.UserFactory;
 import com.api.constants.HttpStatus;
 import com.jayway.jsonpath.JsonPath;
 
-public class UserTests {
+import static com.api.Utils.ResponseAssertUtils.assertUserFieldsPresent;
+import static com.api.constants.TestDataConstants.NON_EXISTENT_USER_ID;
+
+public class UserTests extends BaseApiTest {
 
     @Test
     public void createValidUserTest() {
-        Map<String, String> userData = UserFactory.createDefaultUserData();
-        String template = TemplateUtils.loadTemplate(JsonConstant.USER_TEMPLATE_PATH);
-        String requestBody = TemplateUtils.populateTemplate(template, userData);
-        UserService userService = new UserService();
+        String requestBody = getValidUserRequestBody();
         HttpResponse<String> response = userService.createUser(requestBody);
         Assert.assertEquals(response.statusCode(), HttpStatus.CREATED.getCode(), "Expected status 201 when creating a valid user");
     }
 
     @Test
-    public void getAllUsersTest() {
-        UserService userService = new UserService();
-        HttpResponse<String> response = userService.getAllUsers();
-        Assert.assertEquals(response.statusCode(), HttpStatus.OK.getCode(), "Expected status 200 when retrieving all users");
-        String responseBody = response.body();
-        Assert.assertTrue(responseBody.contains("name"), "Response does not contain 'name'");
-        Assert.assertTrue(responseBody.contains("email"), "Response does not contain 'email'");
-        Assert.assertTrue(responseBody.contains("gender"), "Response does not contain 'gender'");
-        Assert.assertTrue(responseBody.contains("status"), "Response does not contain 'status'");
+    public void createUserWithInvalidDataTest() {
+        String requestBody = getInvalidUserRequestBody();
+        HttpResponse<String> response = userService.createUser(requestBody);
+        Assert.assertEquals(response.statusCode(), HttpStatus.UNPROCESSABLE_ENTITY.getCode(), "Expected status 422 when sending invalid user data");
+    }
+
+    @Test
+    public void createUserWithExistingEmailTest() {
```

Comment:
We could improve maintainability and reduce duplication by using a @DataProvider for our user creation tests (valid, invalid, existing email, etc.).